Segmentation fault upon startup

### Description

When invoked as `containerd -l debug`:

```
DEBU[2022-12-13T19:56:57.779762007Z] Loaded container {Metadata:{ID:84ddf519c8e789676ad7212a6ec91b62f3ca1f031f9e32de5adc61f9a3c6ff45 Name: [...]
DEBU[2022-12-13T19:56:57.887144088Z] Start writing stream "stdout" to log file "/var/log/pods/.../0.log"
DEBU[2022-12-13T19:56:57.887180699Z] Start writing stream "stderr" to log file "/var/log/pods/.../0.log"
DEBU[2022-12-13T19:56:57.888990424Z] Loaded container {Metadata:{ID:85dc0fae6b555b36f5d2ed2920b575c5ddb6b95bdcb0d0535902bebca3b5045d Name: [...]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5639e286cf7d]

goroutine 1839 [running]:
github.com/containerd/containerd/pkg/cri/store/container.deepCopyOf({0x0, 0x17306ebcfa984dc3, 0x0, 0x0, 0x0, {0x0, 0x0}, {0x0, 0x0}, 0x0, ...})
	/go/src/github.com/containerd/containerd/pkg/cri/store/container/status.go:226 +0x2dd
github.com/containerd/containerd/pkg/cri/store/container.(*statusStorage).Get(0xc000c07500?)
	/go/src/github.com/containerd/containerd/pkg/cri/store/container/status.go:210 +0x145
github.com/containerd/containerd/pkg/cri/store/container.WithStatus.func1(0xc00106a840)
	/go/src/github.com/containerd/containerd/pkg/cri/store/container/container.go:81 +0xbc
github.com/containerd/containerd/pkg/cri/store/container.NewContainer({{0xc0030e6700, 0x40}, {0xc000c07680, 0x71}, {0xc0030e6740, 0x40}, 0xc000380c40, {0xc0010a85f0, 0x47}, {0xc0027cd170, ...}, ...}, ...)
	/go/src/github.com/containerd/containerd/pkg/cri/store/container/container.go:96 +0x1cc
github.com/containerd/containerd/pkg/cri/server.(*criService).loadContainer(0xc0000f5400, {0x5639e34c9078?, 0xc0006dcbd0?}, {0x5639e34d0310, 0xc00368e300})
	/go/src/github.com/containerd/containerd/pkg/cri/server/restart.go:321 +0xb87
github.com/containerd/containerd/pkg/cri/server.(*criService).recover(0xc0000f5400, {0x5639e34c9078, 0xc0006dcbd0})
	/go/src/github.com/containerd/containerd/pkg/cri/server/restart.go:80 +0x93c
github.com/containerd/containerd/pkg/cri/server.(*criService).Run(0xc0000f5400)
	/go/src/github.com/containerd/containerd/pkg/cri/server/service.go:208 +0x188
github.com/containerd/containerd/pkg/cri.initCRIService.func1()
	/go/src/github.com/containerd/containerd/pkg/cri/cri.go:106 +0x4c
created by github.com/containerd/containerd/pkg/cri.initCRIService
	/go/src/github.com/containerd/containerd/pkg/cri/cri.go:105 +0x812
```

The segfault occurs after the state recovery portion of the boot process:
```
INFO[2022-12-13T20:14:49.465738023Z] Start subscribing containerd event
INFO[2022-12-13T20:14:49.465831324Z] Start recovering state
INFO[2022-12-13T20:14:49.465791777Z] serving...                                    address=/run/containerd/containerd.sock.ttrpc
INFO[2022-12-13T20:14:49.465899907Z] serving...                                    address=/run/containerd/containerd.sock
INFO[2022-12-13T20:14:49.465918914Z] containerd successfully booted in 0.189234s
panic: runtime error: invalid memory address or nil pointer dereference
```

The segfault does not occur with an empty root/state directory, nor does it occur on Ubuntu 20.04.

### Steps to reproduce the issue

Invoke containerd.

### Describe the results you received and expected

I expected containerd to not segfault.

### What version of containerd are you using?

containerd github.com/containerd/containerd v1.6.12 a05d175400b1145e5e6a735a6710579d181e7fb0

### Any other relevant information

```
runc version 1.1.4
commit: v1.1.4-0-g5fd4c4d1
spec: 1.0.2-dev
go: go1.17.10
libseccomp: 2.5.4
```

```
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04 LTS
Release:	22.04
Codename:	jammy
```

### Show configuration if it is related to CRI plugin.

```
disabled_plugins = [
  "io.containerd.snapshotter.v1.aufs",
  "io.containerd.snapshotter.v1.btrfs",
  "io.containerd.snapshotter.v1.devmapper",
  "io.containerd.snapshotter.v1.native",
  "io.containerd.snapshotter.v1.overlayfs",
  "io.containerd.tracing.processor.v1.otlp",
  "io.containerd.internal.v1.tracing",
  "io.containerd.internal.v1.opt",
]
imports = []
oom_score = 0
plugin_dir = ""
required_plugins = []
root = "/var/lib/containerd"
state = "/run/containerd"
temp = ""
version = 2

[cgroup]
  path = ""

[debug]
  address = ""
  format = ""
  gid = 0
  level = ""
  uid = 0

[grpc]
  address = "/run/containerd/containerd.sock"
  gid = 0
  max_recv_message_size = 16777216
  max_send_message_size = 16777216
  tcp_address = ""
  tcp_tls_ca = ""
  tcp_tls_cert = ""
  tcp_tls_key = ""
  uid = 0

[metrics]
  address = ""
  grpc_histogram = false

[plugins]

  [plugins."io.containerd.gc.v1.scheduler"]
    deletion_threshold = 0
    mutation_threshold = 100
    pause_threshold = 0.02
    schedule_delay = "0s"
    startup_delay = "100ms"

  [plugins."io.containerd.grpc.v1.cri"]
    device_ownership_from_security_context = false
    disable_apparmor = false
    disable_cgroup = false
    disable_hugetlb_controller = false
    disable_proc_mount = false
    disable_tcp_service = true
    enable_selinux = false
    enable_tls_streaming = false
    enable_unprivileged_icmp = false
    enable_unprivileged_ports = false
    ignore_image_defined_volumes = false
    max_concurrent_downloads = 3
    max_container_log_line_size = 16384
    netns_mounts_under_state_dir = false
    restrict_oom_score_adj = false
    sandbox_image = "k8s.gcr.io/pause:3.6"
    selinux_category_range = 1024
    stats_collect_period = 10
    stream_idle_timeout = "4h0m0s"
    stream_server_address = "127.0.0.1"
    stream_server_port = "0"
    systemd_cgroup = false
    tolerate_missing_hugetlb_controller = true
    unset_seccomp_profile = ""

    [plugins."io.containerd.grpc.v1.cri".cni]
      bin_dir = "/opt/cni/bin"
      conf_dir = "/etc/cni/net.d"
      conf_template = ""
      ip_pref = "ipv6"
      max_conf_num = 1

    [plugins."io.containerd.grpc.v1.cri".containerd]
      default_runtime_name = "runc"
      disable_snapshot_annotations = true
      discard_unpacked_layers = false
      ignore_rdt_not_enabled_errors = false
      no_pivot = false
      snapshotter = "zfs"

      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]

        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
          base_runtime_spec = ""
          cni_conf_dir = ""
          cni_max_conf_num = 0
          container_annotations = []
          pod_annotations = []
          privileged_without_host_devices = false
          runtime_engine = ""
          runtime_path = ""
          runtime_root = ""
          runtime_type = "io.containerd.runc.v2"

          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
            BinaryName = ""
            CriuImagePath = ""
            CriuPath = ""
            CriuWorkPath = ""
            IoGid = 0
            IoUid = 0
            NoNewKeyring = false
            NoPivotRoot = false
            Root = ""
            ShimCgroup = ""
            SystemdCgroup = true

      [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime]
        base_runtime_spec = ""
        cni_conf_dir = ""
        cni_max_conf_num = 0
        container_annotations = []
        pod_annotations = []
        privileged_without_host_devices = false
        runtime_engine = ""
        runtime_path = ""
        runtime_root = ""
        runtime_type = ""

        [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime.options]

    [plugins."io.containerd.grpc.v1.cri".image_decryption]
      key_model = "node"

    [plugins."io.containerd.grpc.v1.cri".registry]
      config_path = "/etc/containerd/certs.d"

  [plugins."io.containerd.internal.v1.opt"]
    path = "/opt/containerd"

  [plugins."io.containerd.internal.v1.restart"]
    interval = "10s"

  [plugins."io.containerd.metadata.v1.bolt"]
    content_sharing_policy = "shared"

  [plugins."io.containerd.monitor.v1.cgroups"]
    no_prometheus = false

  [plugins."io.containerd.runtime.v2.task"]
    platforms = ["linux/amd64"]
    sched_core = false

  [plugins."io.containerd.service.v1.diff-service"]
    default = ["walking"]

  [plugins."io.containerd.service.v1.tasks-service"]
    rdt_config_file = ""

  [plugins."io.containerd.snapshotter.v1.zfs"]
    root_path = ""

[proxy_plugins]

[stream_processors]

  [stream_processors."io.containerd.ocicrypt.decoder.v1.tar"]
    accepts = ["application/vnd.oci.image.layer.v1.tar+encrypted"]
    args = ["--decryption-keys-path", "/etc/containerd/ocicrypt/keys"]
    env = ["OCICRYPT_KEYPROVIDER_CONFIG=/etc/containerd/ocicrypt/ocicrypt_keyprovider.conf"]
    path = "ctd-decoder"
    returns = "application/vnd.oci.image.layer.v1.tar"

  [stream_processors."io.containerd.ocicrypt.decoder.v1.tar.gzip"]
    accepts = ["application/vnd.oci.image.layer.v1.tar+gzip+encrypted"]
    args = ["--decryption-keys-path", "/etc/containerd/ocicrypt/keys"]
    env = ["OCICRYPT_KEYPROVIDER_CONFIG=/etc/containerd/ocicrypt/ocicrypt_keyprovider.conf"]
    path = "ctd-decoder"
    returns = "application/vnd.oci.image.layer.v1.tar+gzip"

[timeouts]
  "io.containerd.timeout.bolt.open" = "0s"
  "io.containerd.timeout.shim.cleanup" = "5s"
  "io.containerd.timeout.shim.load" = "5s"
  "io.containerd.timeout.shim.shutdown" = "3s"
  "io.containerd.timeout.task.state" = "2s"

[ttrpc]
  address = ""
  gid = 0
  uid = 0
```

--- a/pkg/cri/store/container/status.go
+++ b/pkg/cri/store/container/status.go
@@ -222,10 +222,12 @@ func deepCopyOf(s Status) Status {
 	if s.Resources != nil && s.Resources.Linux != nil {
 		hugepageLimits := make([]*runtime.HugepageLimit, len(s.Resources.Linux.HugepageLimits))
 		for _, l := range s.Resources.Linux.HugepageLimits {
-			hugepageLimits = append(hugepageLimits, &runtime.HugepageLimit{
-				PageSize: l.PageSize,
-				Limit:    l.Limit,
-			})
+			if l != nil {
+				hugepageLimits = append(hugepageLimits, &runtime.HugepageLimit{
+					PageSize: l.PageSize,
+					Limit:    l.Limit,
+				})
+			}
 		}
 		copy.Resources = &runtime.ContainerResources{
 			Linux: &runtime.LinuxContainerResources{

